client/app/components/dashboards/utils.test.js

```diff
@@ -0,0 +1,1015 @@
+import { keepLayoutsOrder } from "./utils";
+
+// Assumptions for now:
+// titles (textboxes) are always on left
+
+describe("keepLayoutsOrder", () => {
+  it("widget moves to left when there is space", () => {
+    /** 
+        full layout looks like this:
+        ------------- -------------
+        |           | |           |
+        | vis1      | |  vis2     |
+        ------------- -------------
+        ------------- -------------
+        |           | |           |
+        | vis3      | |  vis4     |
+        ------------- -------------
+
+        layout with hidden widget 3 look like this:
+
+        ------------- -------------
+        |           | |           |
+        | vis1      | |  vis2     |
+        ------------- -------------
+        -------------
+        |           | 
+        | vis4      | 
+        ------------- 
+        */
```

Comment:
I simply LOVED the having this visualization as the "docstring" of the test. This was very thoughtful and will enable other people to maintain these tests if needed too.